### PR TITLE
progressive reveal for select tags

### PR DESCRIPTION
**Pull request by danielepolencic**
This PR enables `<select>` tags to conditionally display/hide elements based on their selection.

### CODE REVIEW

**Comment by easternbloc** [comment]
👌 It'll need a squash on merge though but LGTM.

**Comment by gavboulton** [comment]
Looking.

I'd prefer fixing up the commits into one meaningful one rather than squash on commit, this way we remove things like 'refactoring and tidying up'.

**Comment by danielepolencic** [comment]
Not a problem.

I can do that just before merging.

**Comment by easternbloc** [comment]
@gavboulton the squash and merge button lets you change the commit message at that point so you shouldn't get that you can call it whatever you like at that point. I think it should just help us slow down the PR ping pong a little

**Comment by gavboulton** [comment]
True, as a merger, it'll stop me moaning about commit messages!

**Comment by easternbloc** [comment]
Does that mean this is mergable @gavboulton or....?

**Comment by gavboulton** [comment]
Not yet @easternbloc, I'm planning to test it with a screen reader and do some cross-browser testing.

**Comment by merlinc** [comment]
@gavboulton Given we didn't know that there'd be screen reader testing and cross browser testing, is there anything we can do to help this get merged in?

**Comment by merlinc** [comment]
Also, this strikes me as somewhere where having contributing guidelines would clarify what is needed for a PR to be deemed acceptable to be reviewed (e.g. new tests that cover the work, screen shots if there is UI, demonstration of screen reader support if appropriate, etc). 

Might make it easier on all sides :)

**Comment by gavboulton** [comment]
Hey @merlinc, we definitely need contributing guidelines. There are issues open for it across our repos and it's something we should address as part of our roadmap. If you guys can test it that would help but if you're desperate for something I'd suggest adding this in at app level.

File: assets/javascript/progressive-reveal.js [hunk 1]
```diff
@@ -30,35 +33,83 @@ function inputClicked(e, target) {
...
-        inputs = document.getElementsByTagName('input');
-        groups = _.groupBy(inputs, 'name');
-        for (var i = 0, num = inputs.length; i < num; i++) {
-            input = inputs[i];
-            if (input.type.match(/radio|checkbox/)) {
-                helpers.once(input, 'progressive-reveal', setupReveal);
-            }
-        }
+        inputs = document.getElementsByTagName('input'),
+        selects = document.getElementsByTagName('select');
```
Comment (gavboulton) [inline assets/javascript/progressive-reveal.js]: Would this DOM traversal be better after the form length check?

File: assets/javascript/progressive-reveal.js [hunk 2]
```diff
@@ -30,35 +33,83 @@ function inputClicked(e, target) {
...
+function setupRevealForSelectElement(select) {
+    var currentOption = select.options[select.selectedIndex],
+        sectionToggleId = currentOption.getAttribute(toggleAttr),
+        sectionToggle = document.getElementById(sectionToggleId);
 
-    if (toggle) {
-        input.setAttribute('aria-controls', toggleId);
-        inputClicked(null, input);
+    if (sectionToggle) {
+        select.setAttribute('aria-controls', sectionToggleId);
```
Comment (gavboulton) [inline assets/javascript/progressive-reveal.js]: Would `aria-controls` be better on the option?

File: assets/javascript/progressive-reveal.js [hunk 3]
```diff
@@ -30,35 +33,83 @@ function inputClicked(e, target) {
...
+}
+
+function selectChangeListener(select) {
+    return function () {
+        var currentOption = select.options[select.selectedIndex],
+            currentSectionToggleId = currentOption.getAttribute(toggleAttr),
+            currentSectionToggle = document.getElementById(currentSectionToggleId);
+
+        _.each(select.options, function (option) {
+            option.setAttribute('aria-expanded', 'false');
```
Comment (gavboulton) [inline assets/javascript/progressive-reveal.js]: Would setting `aria-expanded` only on `options`s with `data-toggle` be better here?

**Comment by gavboulton** [comment]
As a note, we should be avoiding using select lists where possible. A lot of users struggle with them. It's good to think of alternative solutions such as providing the most popular answers in a radio group. Another option would be a typeahead.

Bear in mind that with JavaScript off/disabled the user will see all progressives reveals so context is important.

**Comment by gavboulton** [comment]
@danielepolencic Are you happy to close this now?

**Pull request closed without merging**